DEPENDENCY_POLICY.md

```diff
@@ -0,0 +1,92 @@
+  and `urls` to reference the version. If you need to reference version `X.Y.Z` as `X_Y_Z`, this
+  may appear in a string as `{underscore_version}`, similarly for `X-Y-Z` you can use
+  `{dash_version}`.
+* Provide accurate entries for `use_category`. Please think carefully about whether there are data
+  or control plane implications of the dependency.
+* CPEs are compulsory for all dependencies that are not purely build/test.
+  [CPEs](https://en.wikipedia.org/wiki/Common_Platform_Enumeration) provide metadata that allow us
+  to correlate with related CVEs in dashboards and other tooling, and also provide a machine
+  consumable join key. You can consult the latest [CPE
+  dictionary](https://nvd.nist.gov/products/cpe) to find a CPE for a dependency.`"N/A"` should only
+  be used if no CPE for the project is available in the CPE database. CPEs should be _versionless_
+  with a `:*` suffix, since the version can be computed from `version`.
+
+When build or test code references Python modules, they should be imported via `pip3_import` in
+[bazel/repositories_extra.bzl](bazel/repositories_extra.bzl). Python modules should not be listed in
+`repository_locations.bzl` entries. `requirements.txt` files for Python dependencies must pin to
+exact versions, e.g. `PyYAML==5.3.1` and ideally also include a [SHA256
+checksum](https://davidwalsh.name/hashin).
+
+Pure developer tooling and documentation builds may reference Python via standalone
+`requirements.txt`, following the above policy.
+
+## New external dependencies
+
+* Any new dependency on the Envoy data or control plane that impacts Envoy core (i.e. is not
+  specific to a single non-core extension) must be cleared with the OSS Envoy security team, please
+  file an issue and tag
+  [@envoyproxy/security-team](https://github.com/orgs/envoyproxy/teams/security-team). While policy
+  is still [evolving](robust_to_untrusted_downstream_and_upstream), criteria that will be used in
+  evaluation include:
+  * Does the project have release versions? How often do releases happen?
+  * Does the project have a security vulnerability disclosure process and contact details?
+  * Does the project have effective governance, e.g. multiple maintainers, a governance policy?
+  * Does the project have a code review culture? Are patches reviewed by independent maintainers
+    prior to merge?
+  * Does the project enable mandatory GitHub 2FA for contributors?
+  * Does the project have evidence of high test coverage, fuzzing, static analysis (e.g. CodeQL),
+    etc.?
+
+* Dependencies for extensions that are tagged as `robust_to_untrusted_downstream` or
+  `robust_to_untrusted_downstream_and_upstream` should be sensitive to the same set of concerns
+  as the core data plane.
+
+## Maintaining existing dependencies
+
+We rely on community volunteers to help track the latest versions of dependencies. On a best effort
+basis:
+
+* Core Envoy dependencies will be updated by the Envoy maintainers/security team.
```

Comment:
We really only update deps when needed for security reasons correct?  Generally if someone wants an upgrade due to some shiny new feature they have done the work and we review.  Are you suggesting we change that, or think it's worth clarifying?

Reply:
I think we usually update dependencies continuously; @moderation has been responsible for a lot of this, see history at https://github.com/envoyproxy/envoy/commits/master/bazel/repository_locations.bzl.

There has been some high profile upgrades that were not really plausible to do mechanically as they required code changes, e.g. `http-parser`, but these are more the exception than rule.

I reckon it's better to try and stick with latest release version. The rationale for this is:
* Many code improvements are made in projects that can impact security that fall below the threshold of CVE.
* Many projects don't have long-term support branches, e.g. if we were talking Linux kernel, maybe it's fine to live on 4.0.x as you know this will have active support for a significant period of time.
* For folks who operate monorepos (cough), other projects force dependencies to be brought forward in time.

The downside is we have more churn and less burn time for a given dependency version.